eng/common/templates-official/steps/source-build.yml

```diff
@@ -105,6 +105,7 @@ steps:
     Contents: |
       **/*.log
       **/*.binlog
+      artifacts/log/binary-report/**
```

Comment:
Thank you for pointing this out. The existing logic won't work to handle the binary-report files since `Updated*.txt` only applies to `test/` and the binary-report files do not contain any `.log` files (only `Updated<baseline-name>.txt` and `NewBinaries.txt`). I updated the uploaded artifacts in https://github.com/dotnet/installer/pull/18726/commits/b63671b983af8c670c93c653dc504d7420fb3ab1 to include the binary-report and tested this change in [this build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2399958&view=results) (internal link). The upload will have to be unconditioned from `parameters.buildSourceOnly` as part of https://github.com/dotnet/source-build/issues/4090